### DISCUSSION TITLE
Is this way of creating reusable components a bad practice in svelte?

### DISCUSSION THREAD

**Discussion by cleytoncarvalho** [sveltejs/kit#10495] [Q&A]
Hi! I'm creating components like these bellow inspired by shadcn ui components library. But I was wondering if this is a bad practice in Svelte to create components like this and if there is problems related with performance that I should be worried about.

![carbon](https://github.com/sveltejs/kit/assets/7952866/8fc95288-ff42-4232-a444-d09582167c8f)

![carbon-2](https://github.com/sveltejs/kit/assets/7952866/c60c9177-f7de-4b95-9fe8-ba396fe8e9d3)

**Comment by kwangure** [comment]
This feels like the wrong abstraction for components. Might have been better if you just used a seperate anchor and button. YMMV. See [Goodbye Clean Code](https://overreacted.io/goodbye-clean-code/).  Svelte's lint/a11y checks also help you better that way.

Also, see [skeleton.dev](https://www.skeleton.dev) for pointers for theming at tailwind level instead of components and [melte-ui](https://www.melt-ui.com/docs/introduction) has nice patterns about pragmatic unstyled components.

Also here are someone's thoughts about [@apply in Tailwind](https://www.youtube.com/watch?v=QBajvZaWLXs&t=202s). 

Maybe use `grid-cols-[....stuff...]` instead of reaching into your children using `:global()`? YMMV. If your children are components i.e `<Component slot="left-icon"/>` they DOM elemenets may not have the `slot` attribute set.
